src/Microsoft.AspNetCore.Blazor.Build/Core/RuntimeDependenciesResolver.cs

```diff
@@ -85,15 +85,24 @@ void ResolveAssembliesCore()
                 {
                     while (pendingAssemblies.TryPop(out var current))
                     {
-                        if (!visitedAssemblies.ContainsKey(current.Name))
+                        if (!visitedAssemblies.Contains(current.Name))
                         {
+                            visitedAssemblies.Add(current.Name);
+
+                            // Not all references will be resolvable within the Mono BCL, particularly
+                            // when building for server-side Blazor as you will be running on CoreCLR
+                            // and therefore may depend on System.* BCL assemblies that aren't present
+                            // in Mono WebAssembly. Skipping unresolved assemblies here is equivalent
+                            // to passing "--skip-unresolved true" to the Mono linker.
                             var resolved = Resolve(current);
-                            visitedAssemblies[current.Name] = resolved;
-                            Results.Add(resolved);
-                            var references = GetAssemblyReferences(resolved);
-                            foreach (var reference in references)
+                            if (resolved != null)
                             {
-                                pendingAssemblies.Push(reference);
+                                Results.Add(resolved);
+                                var references = GetAssemblyReferences(resolved);
+                                foreach (var reference in references)
+                                {
+                                    pendingAssemblies.Push(reference);
+                                }
```

Comment:
Is it worth logging/reporting these as warnings? My experience with MVC is that being intolerant of missing assemblies has prevented a lot of pain for the team (reflection loading/runtime compilation). I understand in this case it's a valid use case, but this will make it easier for mistakes to creep in.